src/client.ts

```diff
@@ -117,6 +117,23 @@ export class Client {
     }
   }
 
+  /**
+   * Queries the receipt corresponding to a transaction hash
+   *
+   * @param txHash Transaction hash returned by call transaction
+   * @return EvmTxReceipt protobuf
```

Comment:
If this method always returns `EvmTxReceipt` can we just deserialize the result here and now and return that? Or is there some case where the caller will actually want the raw bytes?

Reply:
If the user wants to resend the receipt to somewhere else they might prefer the raw bytes. 
If non-EVM plugins decide to opt-in to the TxReceipt idea, they would probably use anther TxReceipt protobuf message. 
But it would be easy enough to change to return an EvmTxReceipt as well?

Reply:
Lets return `EvmTxReceipt`, later we can probably change the signature to support non-evm plugins like so:
```typescript
async getTxReceiptAsync<T extends EvmTxReceipt | TxReceipt>(txHash: Uint8Array, result: T): Promise<T | null>
```